scene/gui/rich_text_label.cpp

```diff
@@ -5562,6 +5562,24 @@ int RichTextLabel::get_line_count() const {
 	return line_count;
 }
 
+Vector2i RichTextLabel::get_line_range(int p_line) {
+	const_cast<RichTextLabel *>(this)->_validate_line_caches();
+
+	int line_count = 0;
+	int to_line = main->first_invalid_line.load();
+	for (int i = 0; i < to_line; i++) {
+		MutexLock lock(main->lines[i].text_buf->get_mutex());
+		int lc = main->lines[i].text_buf->get_line_count();
+
+		if (p_line < line_count + lc) {
+			return main->lines[i].text_buf->get_line_range(p_line - line_count);
```

Comment:
This will return a range within paragraph, so for example for the following text:

```
Test long paragraph test test test.
Test long paragraph test test test.

```
wrapped line this:
<img width="160" alt="Screenshot 2024-12-15 at 20 00 19" src="https://github.com/user-attachments/assets/1110086f-65cb-451e-904b-376d0b415419" />

it will return:
```
0 -> (0, 24)
1 -> (25, 36)
2 -> (0, 24)
3 -> (25, 36)
4 -> (0, 0)
```

In most cases, you probably want global range instead:
```suggestion
			return main->lines[i].text_buf->get_line_range(p_line - line_count) + Vector2i(main->lines[i].char_offset, main->lines[i].char_offset);
```

with this change, it will return:
```
0 -> (0, 24)
1 -> (25, 36)
2 -> (36, 60)
3 -> (61, 72)
4 -> (72, 72)
```